code/game/verbs/suicide.dm

```diff
@@ -38,7 +38,10 @@
 			return 1
 
 /mob/living/proc/handle_suicide_bomb_cause()
+	var/old_canmove = canmove
+	canmove = FALSE // Prevent moving away and ruining this
```

Comment:
I don't know about this one. `canmove` will be changed by many things, such as resting and standing up. Maybe you want to stun them instead?

Reply:
what would be best? set stun to INFINITY or a new locking category? looking through update canmove right now